### PR TITLE
PlatformPkgGcc.fdf, PlatformPkg.dec: add arbitrary files as a raw binary objects

### PR DESCRIPTION
**Pull request by krystian-hebel**
- FLASH_REGION_BINARY_FIRMWARE includes a binary file as-is, this way
  position can be precisely definied
- FLASH_REGION_BINARY_FIRMWARE2 embeds file in separate FV, there are some
  headers, so definied position isn't quite accurate

We should choose just one of these for final release

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

### CODE REVIEW

**Comment by miczyg1** [comment]
@krystian-hebel rather the first approach is appropriate. Are there any advantages of embedding EFI file in a separate FV?

**Comment by krystian-hebel** [comment]
@miczyg1 in the second approach they can be added/removed/edited by tools like UEFITool, as well as compressed (see FVMAIN_COMPACT in .fdf file), but for compression adding them to FVMAIN would be enough.

EDIT:
Also their GUIDs and sizes can be retrieved from FV.

File: Vlv2TbltDevicePkg/PlatformPkgGcc.fdf [hunk 1]
```diff
@@ -152,6 +158,20 @@ $(FLASH_REGION_FVMAIN_OFFSET)|$(FLASH_REGION_FVMAIN_SIZE)
 gPlatformModuleTokenSpaceGuid.PcdFlashFvMainBase|gPlatformModuleTokenSpaceGuid.PcdFlashFvMainSize
 FV = FVMAIN_COMPACT
 
+  #
+  # Binary Firmware
+  #
+$(FLASH_REGION_BINARY_FIRMWARE_OFFSET)|$(FLASH_REGION_BINARY_FIRMWARE_SIZE)
+gPlatformModuleTokenSpaceGuid.PcdFlashBinFwBase|gPlatformModuleTokenSpaceGuid.PcdFlashBinFwSize
```
Comment (miczyg1) [inline Vlv2TbltDevicePkg/PlatformPkgGcc.fdf:162]: @krystian-hebel I can see the file in firmware image, but it is shown as a pad file. Is there any possibility to add some description for this file to find it easier?

Comment (krystian-hebel) [reply Vlv2TbltDevicePkg/PlatformPkgGcc.fdf:162]: @miczyg1 not using binary file, I think, they're just blobs. In FV a file can have more than one section, in the `User Interface section` is usually a name for binaries, although I didn't research a specification for this.

Comment (miczyg1) [reply Vlv2TbltDevicePkg/PlatformPkgGcc.fdf:162]: @krystian-hebel it is just harder for for typical user to find the file in current form. However the file is at desired offset, this is the only information about the file we have to search for it for example in UEFI tool. Would be good if we could have something like a RESERVED type of file.

**Comment by miczyg1** [comment]
@krystian-hebel I have checked and both ways work. However we still can replace binary embedded by 1st method using `Replace as is`. I wonder what will be needed next when we will move to using trampoline. Would be 2nd approach more easy to handle?
@pietrushnic FYI

**Comment by pietrushnic** [comment]
@krystian-hebel please address @miczyg1 question.

**Comment by krystian-hebel** [comment]
@miczyg1 I'm not sure. 1st one can be just loaded from memory, while 2nd needs to be accessed by FirmwareVolume2 (name might by wrong, I'm writing from memory and it's late), but perhaps RTNic needs a device path with some kind of file system. I'm opting for 1st option, as long as we have enough space, but 2nd may be needed as a backup plan.

**Comment by miczyg1** [comment]
@krystian-hebel if it is easier to handle binaries via memory addresses and You plan to use 1st approach in further development, let's stick to that one method. The second option may land to our documentation to not loose it if needed

**Comment by miczyg1** [comment]
@krystian-hebel please change PR description too

**Comment by miczyg1** [comment]
@pietrushnic @krystian-hebel i have no objections to the change. So I am moving it to a branch with more descriptive name: https://github.com/3mdeb/edk2-platforms/tree/arbitrary_binary so there will be no conflicts with other features

**Pull request closed without merging**